src/Uno.UI/UI/Xaml/Controls/TextBox/TextBox.Android.cs

```diff
@@ -371,9 +371,38 @@ private void UpdateInputScope(InputScope inputScope)
 
 		partial void OnTextWrappingChangedPartial(DependencyPropertyChangedEventArgs e)
 		{
-			//TODO : see bug #8178
+			if (e.NewValue is TextWrapping textWrapping && !string.IsNullOrEmpty(Text) &&
+				(textWrapping is TextWrapping.Wrap || textWrapping is TextWrapping.WrapWholeWords))
+			{
+				if (_textBoxView != null)
+				{
+					//store  the value configured on the wrapper _textBoxView to restore if user changes from Wrap to UnWrap
+					_isSingleLine = _textBoxView.IsSingleLine;
```

Comment:
Why is it necessary to store the "old" value instead of just directly setting true/false?

Reply:
You should be able to invoke `_textBoxView?.SetSingleLine(false);` directly, without storing the `IsSingleLine` as the only interactions for `TextBoxView` come from `TextBox`.